pkg/autostart/autostart.go

```diff
@@ -0,0 +1,137 @@
+// Package autostart manage start at login unit files for darwin/linux
+package autostart
+
+import (
+	_ "embed"
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/lima-vm/lima/pkg/textutil"
+)
+
+//go:embed lima-vm@INSTANCE.service
+var systemdTemplate string
+
+//go:embed io.lima-vm.autostart.INSTANCE.plist
+var launchdTemplate string
+
+// CreateStartAtLoginEntry respect host OS arch and create unit file
+func CreateStartAtLoginEntry(hostOS, instName, workDir string) error {
+	unitPath := GetStartAtLoginEntryPath(hostOS, instName)
+	if _, err := os.Stat(unitPath); !errors.Is(err, os.ErrNotExist) {
+		return err
+	}
+	tmpl, err := renderTemplate(hostOS, workDir, instName, os.Executable)
+	if err != nil {
+		return err
+	}
+	if err := os.WriteFile(unitPath, tmpl, 0o644); err != nil {
+		// Early return if error is unexpected
+		if !errors.Is(err, os.ErrExist) && !errors.Is(err, os.ErrNotExist) {
+			return err
+		}
+		// Unit file exists, return nil
+		if errors.Is(err, os.ErrExist) {
+			return nil
+		}
+		// .config/systemd folder might be unavailable
+		if errors.Is(err, os.ErrNotExist) && hostOS == "linux" {
+			if err := os.MkdirAll(filepath.Dir(unitPath), os.ModePerm); err != nil {
+				return err
+			}
+			// directory created try to create unit file
+			if err := os.WriteFile(unitPath, tmpl, 0o644); err != nil {
+				return err
+			}
+		}
+	}
+	return enableDisableService("enable", hostOS, GetStartAtLoginEntryPath(hostOS, instName))
+}
+
+// DeleteStartAtLoginEntry respect host OS arch and delete unit file
+// return true, nil if unit file has been deleted
+func DeleteStartAtLoginEntry(hostOS, instName string) (bool, error) {
+	unitPath := GetStartAtLoginEntryPath(hostOS, instName)
+	if _, err := os.Stat(unitPath); err != nil {
+		// Skip error log if file not present
+		if errors.Is(err, os.ErrNotExist) {
+			return false, nil
+		}
+		return false, err
+	}
+	if err := enableDisableService("disable", hostOS, GetStartAtLoginEntryPath(hostOS, instName)); err != nil {
+		return false, err
+	}
+	if err := os.Remove(unitPath); err != nil {
+		return false, err
+	}
+	return true, nil
+}
+
+// GetFilePath returns the path to autostart file with respect of host
+func GetStartAtLoginEntryPath(hostOS, instName string) string {
+	var fileTmp string
+	if hostOS == "darwin" { // launchd plist
+		fileTmp = fmt.Sprintf("%s/Library/LaunchAgents/io.lima-vm.autostart.%s.plist", os.Getenv("HOME"), instName)
+	}
+	if hostOS == "linux" { // systemd service
```

Comment:
Can we move these if conditions to separate os specific files.

Reply:
I don't think it make code more readable, there is not so much code to split in separate files.  Code in one file is much easy to maintain.